ext/systemd/puppet.service

```
@@ -11,7 +11,7 @@
 [Unit]
 Description=Puppet agent
 Wants=basic.target
-After=basic.target network.target
+After=basic.target network-online.target
```

Depending on network-online.target is good. I use that for all services I maintain. I don't know if puppet is used on systems without a proper network manager. In those edge cases you won't have `network-online.target`. So we could do:
```suggestion
After=basic.target network.target network-online.target
```

I prefer `Wants=basic.target network.target network-online.target` because IMO enabling the puppet unit without network is useless und jusst triggers errror in the puppet journal. But that change could surprise users. I recommend updating `Wants=` for Puppet 8.